src/components/GiftRanking.tsx

```diff
@@ -105,12 +105,30 @@ const rankingTabs = [
   { key: 'wish', label: '위시로 받은' },
 ];
 
-
 export default function GiftRankingSection() {
-  const [filter, setFilter] = useState('all');
-  const [tab, setTab] = useState('want');
+  const [searchParams, setSearchParams] = useSearchParams();
+  const initGender = searchParams.get('gender') ?? 'all';
+  const initType = searchParams.get('type') ?? 'want';
+
+  const [filter, setFilter] = useState(initGender);
+  const [tab, setTab] = useState(initType);
   const [collapsed, setCollapsed] = useState(true);
 
+  const updateParams = (key: string, value: string) => {
+    const next = new URLSearchParams(searchParams);
+    next.set(key, value);
+    setSearchParams(next, { replace: true });
+  };
+
+  const handleFilter = (key: string) => {
+    setFilter(key);
+    updateParams('gender', key);
+  };
+  const handleTab = (key: string) => {
```

Comment:
아시겠지만 URL 파라미터는 key-value 쌍인데요.
여기서 받는 인자를 key라는 이름보다는 value가 더 어울릴 것 같아요! 나중에 봤을 때 헷갈릴 수도 있으니까요 😀
```suggestion
  const handleTab = (value: string) => {
```